Formula/xgboost.rb

```diff
@@ -13,10 +14,19 @@ class Xgboost < Formula
   end
 
   depends_on "cmake" => :build
+  depends_on "libomp"
 
   def install
     mkdir "build" do
-      system "cmake", *std_cmake_args, "-DUSE_OPENMP=0", ".."
+      libomp = Formula["libomp"]
+      args = std_cmake_args
+      args << "-DOpenMP_C_FLAGS=\"-Xpreprocessor -fopenmp -I#{libomp.opt_include}\""
+      args << "-DOpenMP_C_LIB_NAMES=omp"
+      args << "-DOpenMP_CXX_FLAGS=\"-Xpreprocessor -fopenmp -I#{libomp.opt_include}\""
+      args << "-DOpenMP_CXX_LIB_NAMES=omp"
+      args << "-DOpenMP_omp_LIBRARY=#{libomp.opt_lib}/libomp.dylib"
```

Comment:
Honestly, CMake should be able to do this automatically nowadays.

Looks like it might be a bug with the superenv:

```
clang++ called with: -Xclang -fopenmp -v -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk -o CMakeFiles/cmTC_29998.dir/OpenMPTryFlag.cpp.o -c /tmp/xgboost-20200424-59892-1wvj0bb/build/CMakeFiles/FindOpenMP/OpenMPTryFlag.cpp
superenv removed:  -fopenmp -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk
```

I'll check up on this.

Reply:
Thanks @Bo98! The additional flags aren't needed outside the Homebrew environment.